### PR TITLE
Add integration tests for batch logging

### PR DESCRIPTION
**Pull request by dsuhinin**

### CODE REVIEW

File: pkg/api/mlflow/dao/models/metric.go [hunk 1]
```diff
@@ -11,6 +11,11 @@ type Metric struct {
 	Iter      int64   `gorm:"index"`
 }
 
+// TableName returns actual table name.
+func (p Metric) TableName() string {
+	return "metrics"
+}
+
```
Comment (jgiannuzzi) [inline pkg/api/mlflow/dao/models/metric.go:18]: I have seen in other PRs that we introduced these `TableName` methods. Out of curiosity, what is the reasoning behind them?

Comment (dsuhinin) [reply pkg/api/mlflow/dao/models/metric.go:18]: ahh, yep. This is `tabler` interface and it needs to run this logic:
```
for _, table := range []schema.Tabler{
		models.Tag{},
		models.Param{},
		models.LatestMetric{},
		models.Metric{},
		models.Run{},
		models.ExperimentTag{},
		models.Experiment{},
	} {
		if err := f.db.Session(&gorm.Session{AllowGlobalUpdate: true}).Delete(table).Error; err != nil {
			return errors.Wrapf(err, "error deleting data from %s table", table.TableName())
		}
	}
```

File: pkg/api/mlflow/dao/repositories/metric.go [hunk 2]
```diff
@@ -23,11 +23,15 @@ type MetricRepositoryProvider interface {
...
 	CreateBatch(ctx context.Context, run *models.Run, batchSize int, params []models.Metric) error
+	// GetMetricByKey returns metric by provided key.
+	GetMetricByKey(ctx context.Context, key string) (*models.Metric, error)
 	// GetMetricHistories returns metric histories by request parameters.
 	GetMetricHistories(
 		ctx context.Context,
 		experimentIDs []string, runIDs []string, metricKeys []string, viewType request.ViewType, limit int32,
 	) (*sql.Rows, func(*sql.Rows, interface{}) error, error)
+	// GetLatestMetricByKey returns the latest metric by provided key.
+	GetLatestMetricByKey(ctx context.Context, key string) (*models.LatestMetric, error)
```
[the remark this answers is not in the document]
Comment (dsuhinin) [reply pkg/api/mlflow/dao/repositories/metric.go]: sorry, only one probably we need.

[the remark this answers is not in the document]
Comment (dsuhinin) [reply pkg/api/mlflow/dao/repositories/metric.go]: just deleted `GetMetricByKey`

File: pkg/api/mlflow/dao/models/metric.go (hunk 1 above)
Comment (jgiannuzzi) [reply pkg/api/mlflow/dao/models/metric.go:18]: hmmm, I am pretty sure that there is a default naming method in Gorm that will give you the name instead

File: pkg/api/mlflow/dao/repositories/metric.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jgiannuzzi) [reply pkg/api/mlflow/dao/repositories/metric.go]: but is the other method only there for the tests and not for actual use in the service?

File: pkg/api/mlflow/dao/models/metric.go (hunk 1 above)
Comment (dsuhinin) [reply pkg/api/mlflow/dao/models/metric.go:18]: it is a pretty common practise. I can't see any problem to follow Gorm standards.

Comment (dsuhinin) [reply pkg/api/mlflow/dao/models/metric.go:18]: gorm could generate it somewhere internally - yes, but I prefer to control it and use for logging as well like:
```
return errors.Wrapf(err, "error deleting data from %s table", table.TableName())
```

File: pkg/api/mlflow/dao/repositories/metric.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dsuhinin) [reply pkg/api/mlflow/dao/repositories/metric.go]: it is only for test. do you have any objections or concerns to do not have it?

[the remark this answers is not in the document]
Comment (dsuhinin) [reply pkg/api/mlflow/dao/repositories/metric.go]: ok, removed it from `repo` now.

File: pkg/api/mlflow/dao/models/metric.go (hunk 1 above)
Comment (jgiannuzzi) [reply pkg/api/mlflow/dao/models/metric.go:18]: so this is only used for logging in tests — I would argue that we shouldn't define all of this code just for that, especially as the `err` returned by Gorm is already "no such table: runs", so simply having `return errors.Wrapf(err, "error unloading fixtures")` would give us all the information we need

Comment (dsuhinin) [reply pkg/api/mlflow/dao/models/metric.go:18]: you are strong fighter =). ok let me check.

Comment (dsuhinin) [reply pkg/api/mlflow/dao/models/metric.go:18]: done.

**Review by jgiannuzzi — APPROVED** [review]

**Pull request merged**